src/Illuminate/Support/Testing/Fakes/ExceptionHandlerFake.php

```diff
@@ -0,0 +1,267 @@
+                    && $exception($e) === true,
+            ), $message,
+        );
+    }
+
+    /**
+     * Assert the number of exceptions that have been reported.
+     *
+     * @param  int  $count
+     * @return void
+     */
+    public function assertReportedCount(int $count)
+    {
+        $total = collect($this->reported)->count();
+
+        PHPUnit::assertSame(
+            $count, $total,
+            "The total number of exceptions reported was {$total} instead of {$count}."
+        );
+    }
+
+    /**
+     * Assert if an exception of the given type has not been reported.
+     *
+     * @param  \Closure|string  $exception
+     * @return void
+     */
+    public function assertNotReported(Closure|string $exception)
+    {
+        try {
+            $this->assertReported($exception);
+        } catch (ExpectationFailedException $e) {
+            return;
+        }
+
+        throw new ExpectationFailedException(sprintf(
+            'The expected [%s] exception was not reported.',
+            is_string($exception) ? $exception : $this->firstClosureParameterType($exception)
+        ));
+    }
+
+    /**
+     * Report or log an exception.
+     *
+     * @param  \Throwable  $e
+     * @return void
+     */
+    public function report($e)
+    {
+        if ($this->shouldReport($e) && $this->isFakedException($e)) {
+            $this->reported[] = $e;
+
+            if ($this->throwOnReport) {
+                throw $e;
+            }
+        }
+
+        $this->handler->report($e);
+    }
+
+    /**
+     * Determine if the exception should be reported.
+     *
+     * @param  \Throwable  $e
+     * @return bool
+     */
+    public function shouldReport($e)
+    {
+        return $this->runningWithoutExceptionHandling() || $this->handler->shouldReport($e);
+    }
+
+    /**
+     * Render an exception into an HTTP response.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Throwable  $e
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function render($request, $e)
+    {
+        return $this->handler->render($request, $e);
+    }
+
+    /**
+     * Render an exception to the console.
+     *
+     * @param  \Symfony\Component\Console\Output\OutputInterface  $output
+     * @param  \Throwable  $e
+     * @return void
+     */
+    public function renderForConsole($output, Throwable $e)
+    {
+        $this->handler->renderForConsole($output, $e);
+    }
+
+    /**
+     * Throw exceptions when they are reported.
+     *
+     * @return $this
+     */
+    public function throwOnReport()
+    {
+        $this->throwOnReport = true;
+
+        return $this;
+    }
+
+    /**
+     * Throw all of the exceptions that have been reported.
+     *
+     * @return $this
+     *
+     * @throws \Throwable
+     */
+    public function throwReported()
+    {
+        foreach ($this->reported as $e) {
+            throw $e;
+        }
+
+        return $this;
+    }
```

Comment:
What is the usecase for calling `throwReported` instead of `throwOnReport`? It feels like these achieve the same thing but it just depends _where_ in the test you call them.

It feels like `throwReported` may be a redundant API and it might be cleaner if we only include one way of doing things, i.e., remove `throwReported` in favor of `throwOnReport`?

If we do decide to keep it, we might want to consider a name change. It will also only ever throw the first exception to `throwFirstReported`.